test/avocado/selenium-kdump.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+import os
+import sys
+import time
+machine_test_dir = os.path.dirname(os.path.realpath(__file__))
+if machine_test_dir not in sys.path:
+    sys.path.insert(1, machine_test_dir)
+
+from selenium.webdriver.support.select import Select
+from testlib_avocado.seleniumlib import SeleniumTest, clickable, visible
+
+
+class TestKdump(SeleniumTest):
+    """
+    :avocado: enable
+    """
+    share = "/nfsfileshare"
+    exporttab = """
+{}    *(rw,sync,no_root_squash)
+    """.format(share)
+    export_file = '/etc/exports'
+
+
+    def requirements(self):
+        execs = [
+            # start NFS server
+            "sudo yum install -y nfs-utils || true",
+            "sudo mkdir -p {}".format(self.share),
+            "sudo chmod a+rwx {}".format(self.share),
+            "echo '{}' | sudo tee {}".format(self.exporttab, self.export_file),
+            "sudo systemctl start nfs-server rpcbind",
+            "sudo exportfs -r",
+            # TODO: these two lines are workaround, because it is unable otherwise create initrd for nfs
+            "sudo mkdir -p {}/var/crash".format(self.share),
+            "sudo mount localhost:{} /var/crash".format(self.share),
+        ]
+        for exe in execs:
+            self.machine.execute(exe)
+
+    def requirements_cleanup(self):
+        execs = [
+            "sudo umount /var/crash",
+            "sudo systemctl stop nfs-server rpcbind",
+            "echo '' | sudo tee {}".format(self.export_file),
+            "sudo rm -rf {}".format(self.share),
+        ]
+        for exe in execs:
+            self.machine.execute(exe)
+
+    def setUp(self):
+        super().setUp()
+        self.login()
+        self.requirements()
+        self.click(self.wait_link("Kernel Dump"))
+        self.wait_frame("localhost/kdump")
+        self.base_element = self.wait_id("app", jscheck=True)
+
+    def tearDown(self):
+        pass
+        self.requirements_cleanup()
+        super().tearDown()
+
+    def testKdumpBasePage(self):
+        self.wait_text("Crash dump location", cond=visible)
+
+    def check_type(self, old_text, new_text, selector_item, textbox_dict):
+        self.wait_text("Service is running", cond=visible)
```

Comment:
It times out here. Shouldn't it click on the on/off button first to actually enable it?

Reply:
Hi Martin, yes.
you are right. I've missed that basic RHEL7 and RHEL8 images what I've used had enabled kdump, but fedora one does not have kernel param to enable kdump.
Thanks for finding it. I've also discovered it.
Stranger is that bug: ``write could not complete without blocking`` what caused bear in my mind, because it was not able then do screenshots. I will prolong period for page loading.